### PR TITLE
make butil::ScopedVector<T> support std::initializer_list

### PR DESCRIPTION
**Pull request by lrita**

### CODE REVIEW

File: src/butil/memory/scoped_vector.h [hunk 1]
```diff
@@ -38,6 +41,9 @@ class ScopedVector {
   ScopedVector() {}
   ~ScopedVector() { clear(); }
   ScopedVector(RValue other) { swap(*other.object); }
+#if __cplusplus >= 201103L  // < C++11
```
Comment (gydong) [inline src/butil/memory/scoped_vector.h:44]: >= c++11 
  or just delete this comment since the statement is self-descriptive enough.

Comment (lrita) [reply src/butil/memory/scoped_vector.h]: fix the comment

**Comment by gydong** [comment]
LGTM

**Comment by lrita** [comment]
Hi @jamesge, would you have time to review this pr.

File: src/butil/memory/scoped_vector.h [hunk 2]
```diff
@@ -6,6 +6,9 @@
 #define BUTIL_MEMORY_SCOPED_VECTOR_H_
 
 #include <vector>
+#if __cplusplus >= 201103L  // >= C++11
```
Comment (jamesge) [inline src/butil/memory/scoped_vector.h:9]: no need to add this, c++11 is enabled by default

Comment (lrita) [reply src/butil/memory/scoped_vector.h]: I have fixed it.

**Pull request merged**